dotnet-test-samples/async-architectures/async-lambda-sqs/tests/S3Notifications.integrationTests/Fixtures/SqsTestFixture.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Amazon;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+
+namespace S3Notifications.integrationTests.Fixtures
+{
+    public class SqsTestFixture : IDisposable
+    {   
+        internal IAmazonSQS SqsClient { get; }
+        internal string QueueUrl { get; }
+
+        private static readonly RegionEndpoint Region = RegionEndpoint.USEast1;
+        private const string QueueNamePrefix = "test-s3-queue";
+
+        public SqsTestFixture()
+        {
+            SqsClient = new AmazonSQSClient(Region);
+
+            var queueName = $"{QueueNamePrefix}-{Guid.NewGuid()}";
+
+            var createQueueResponse = SqsClient.CreateQueueAsync(queueName).Result;
+            QueueUrl = createQueueResponse.QueueUrl;
+        }
+
+        public void Dispose()
+        {
+            SqsClient.DeleteQueueAsync(QueueUrl).Wait();
+            SqsClient.Dispose();
+        }
+        
+        public async Task<Message> GetNextMessage()

```

Comment:
Same as above here, how would you check you're getting the correct message

Reply:
Since I'm creating the queue specifically for the tests I can assume only my test runs use this queue... Not sure if we need to accommodate to parallel runs - so I would expect to get exactly the message I need, or fail the test

Reply:
Makes sense :)